pkg/fanal/analyzer/language/java/pom/pom.go

```diff
@@ -23,11 +25,23 @@ const version = 1
 type pomAnalyzer struct{}
 
 func (a pomAnalyzer) Analyze(_ context.Context, input analyzer.AnalysisInput) (*analyzer.AnalysisResult, error) {
-	p := pom.NewParser(filepath.Join(input.Dir, input.FilePath), pom.WithOffline(input.Options.Offline))
+	filePath := filepath.Join(input.Dir, input.FilePath)
+	p := pom.NewParser(filePath, pom.WithOffline(input.Options.Offline))
 	res, err := language.Analyze(types.Pom, input.FilePath, input.Content, p)
 	if err != nil {
 		return nil, xerrors.Errorf("%s parse error: %w", input.FilePath, err)
 	}
+
+	// Mark integration test pom files for `maven-invoker-plugin` as Dev to skip them by default.
+	if isIntegrationTestDir(filePath) {
+		for i := range res.Applications {
+			res.Applications[i].Libraries = lo.Map(res.Applications[i].Libraries, func(lib types.Package, _ int) types.Package {
```

Comment:
It is no big deal, but `lo.Map` generates another map, and it is better just to overwrite the `Dev` field from the memory perspective.

Reply:
Fixed in https://github.com/aquasecurity/trivy/pull/6213/commits/b86dff88dcc68c8d2b85790694793a3fd06fc2c8